test/client-request.js

```diff
@@ -97,3 +99,85 @@ test('trailers', (t) => {
       })
   })
 })
+
+test('destroy socket abruptly', async (t) => {
+  t.plan(2)
+
+  const server = net.createServer((socket) => {
+    const lines = [
+      'HTTP/1.1 200 OK',
+      'Date: Sat, 09 Oct 2010 14:28:02 GMT',
+      'Connection: close',
+      '',
+      'the body'
+    ]
+    socket.end(lines.join('\r\n'))
+
+    // Unfortunately calling destroy synchronously might get us flaky results,
+    // therefore we delay it to the next event loop run.
+    socket.destroy()
+  })
+  t.teardown(server.close.bind(server))
+
+  await promisify(server.listen.bind(server))(0)
+  const client = new Client(`http://localhost:${server.address().port}`)
+  t.teardown(client.close.bind(client))
+
+  const { statusCode, body } = await client.request({
+    path: '/',
+    method: 'GET'
+  })
+
+  t.equal(statusCode, 200)
+
+  body.setEncoding('utf8')
+
+  let actual = ''
+
+  for await (const chunk of body) {
+    actual += chunk
+  }
+
+  t.equal(actual, 'the body')
+})
+
+test('destroy socket abruptly with keep-alive', async (t) => {
```

Comment:
Shouldn't this test error?

Reply:
We are not erroring right now because shouldKeepAlive is coming as false from llhttp.

Reply:
> shouldKeepAlive is coming as false from llhttp.

Why? That sounds wierd?

Reply:
This `shouldKeepAlive` https://github.com/nodejs/undici/blob/e61c2525a73f137a519c43cd6ada7c0334cdee04/lib/client.js#L402 is coming as `0`. Maybe @indutny can help.

Reply:
I've found the problem, content-length was missing. pushing a fix.